bls-remote-signer/backend/src/lib.rs

```diff
@@ -96,15 +97,19 @@ impl<T: Storage> Backend<T> {
     /// Computes the public key from the retrieved `secret_key` and compares it
     /// with the given `public_key` parameter.
     fn validate_bls_pair(public_key: &str, secret_key: &str) -> Result<SecretKey, BackendError> {
+        // TODO
+        // Can I do this in one error message?
```

Comment:
The two different error types make it a challenge, I don't think you can get a whole lot better than this. 

This is the best I could do, I wouldn't really suggest it though:

```rust
let secret_key = hex_string_to_bytes(&secret_key)
    .map_err(|e| e.to_string())
    .and_then(|bytes| {
        SecretKey::deserialize(&secret_key).map_err(|e| format!("{:?}", e))
    })
    .map_err(|e| {
        BackendError::InvalidSecretKey(format!(
            "public key: {}; {}",
            public_key, e
        ))
    })?;
```

Reply:
Nice! I got this little contraption. On the fence of putting it, as it uses three `try` macros. But let's keep it for archive purposes :slightly_smiling_face: 

```rust
let deserialize = |sk: &str| -> Result<SecretKey, String> {
    let sk = hex_string_to_bytes(&sk).map_err(|e| e.to_string())?;
    Ok(SecretKey::deserialize(&sk).map_err(|e| format!("{:?}", e))?)
};

let secret_key: SecretKey = deserialize(secret_key).map_err(|e| {
    BackendError::InvalidSecretKey(format!("public_key: {}; {}", public_key, e))
})?;
```